shared/cloud-gateway/develop/_send-receive-media-streams-c.mdx

```diff
@@ -0,0 +1,540 @@
+
+    ```cpp
+    // Creates a sender for PCM audio
+    agora::agora_refptr<agora::rtc::IAudioPcmDataSender> audioPcmDataSender =
+        factory->createAudioPcmDataSender();
+    if (!audioPcmDataSender) {
+    AG_LOG(ERROR, "Failed to create audio data sender!");
+    return -1;
+    }
+
+    // Creates a sender for YUV video
+    agora::agora_refptr<agora::rtc::IVideoFrameSender> videoFrameSender =
+        factory->createVideoFrameSender();
+    if (!videoFrameSender) {
+    AG_LOG(ERROR, "Failed to create video frame sender!");
+    return -1;
+    }
+
+    // Creates a sender for encoded audio
+    agora::agora_refptr<agora::rtc::IAudioEncodedFrameSender> audioFrameSender =
+        factory->createAudioEncodedFrameSender();
+    if (!audioFrameSender) {
+        AG_LOG(ERROR, "Failed to create audio encoded frame sender!");
+        return -1;
+    }
+    
+    // Creates a sender for encoded video
+    agora::agora_refptr<agora::rtc::IVideoEncodedImageSender> videoEncodedFrameSender =
+        factory->createVideoEncodedImageSender();
+    if (!videoEncodedFrameSender) {
+    AG_LOG(ERROR, "Failed to create encoded video frame sender!");
+    return -1;
+    }
+    ```
+
+3. Create an `ILocalAudioTrack` object and an `ILocalVideoTrack` object, which correspond to the local audio track and local video track to be published to the Agora RTC channel:
+
+    ```c++
+    // Creates a custom audio track that uses a PCM audio stream sender
+    agora::agora_refptr<agora::rtc::ILocalAudioTrack> customAudioTrack =
+        service->createCustomAudioTrack(audioPcmDataSender);
+    if (!customAudioTrack) {
+    AG_LOG(ERROR, "Failed to create audio track!");
+    return -1;
+    }
+
+    // Creates a custom audio track that uses an encoded audio stream sender
+    agora::agora_refptr<agora::rtc::ILocalAudioTrack> customAudioTrack =
+        service->createCustomAudioTrack(audioFrameSender, agora::base::MIX_DISABLED);
+    if (!customAudioTrack) {
+    AG_LOG(ERROR, "Failed to create audio track!");
+    return -1;
+    }
+
+    // Creates a custom video track that uses a YUV video stream sender
+    agora::agora_refptr<agora::rtc::ILocalVideoTrack> customVideoTrack =
+        service->createCustomVideoTrack(videoFrameSender);
+    if (!customVideoTrack) {
+    AG_LOG(ERROR, "Failed to create video track!");
+    return -1;
+    }
+
+    // Creates a custom video track that uses an encoded video stream sender
+    agora::agora_refptr<agora::rtc::ILocalVideoTrack> customVideoTrack =
+        service->createCustomVideoTrack(videoEncodedFrameSender);
+    if (!customVideoTrack) {
+    AG_LOG(ERROR, "Failed to create video track!");
+    return -1;
+    }
+    ```
+
+### Step 2: Send media streams
+
+1. Use `publish` methods in the `ILocalUser` object to publish local audio and video tracks created in the previous step to the Agora RTC channel:
+
+    ```c++
```

Comment:
Code highlighting in the following snippets looks off.

Reply:
Fixed.